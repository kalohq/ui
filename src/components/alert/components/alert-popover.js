/* @flow */
import * as React from 'react';
import {Box} from '../../layout';
import CSSTransitionGroup from 'react-addons-css-transition-group';

type TProps = {
  children: React.Element<*>,
  topOffset?: number,
  zIndex?: number,
};

export default function AlertPopover(props: TProps) {
  const {children, topOffset = 0, zIndex = 999} = props;

  return (
    <CSSTransitionGroup
      transitionName="t-slide-from-right"
      transitionAppear={true}
      transitionAppearTimeout={300}
      transitionEnterTimeout={300}
      transitionLeaveTimeout={300}
      component={Box}
      position="fixed"
      right={32}
      top={topOffset + 16}
      maxWidth={400}
      alignItems="flex-end"
      zIndex={zIndex}
    >
      {React.Children.map(
        children,
        child =>
          child ? (
            <Box marginBottom={8}>
              {React.cloneElement(child, {justifyContent: 'left'})}
            </Box>
          ) : null
      )}
    </CSSTransitionGroup>
  );
}
