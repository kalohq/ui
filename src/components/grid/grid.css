:root {
  --gutter-width: 16px;
  --outer-margin: 32px;
  --gutter-compensation: calc((var(--gutter-width) * 0.5) * -1);
  --half-gutter-width: calc((var(--gutter-width) * 0.5));
}

.container {
  margin-right: auto;
  margin-left: auto;
  width: 100%;
  max-width: 1600px;
}

.container.fluid {
  padding-right: var(--outer-margin, 2rem);
  padding-left: var(--outer-margin, 2rem);
}

.row {
  box-sizing: border-box;
  display: flex;
  flex: 0 1 auto;
  flex-direction: row;
  flex-wrap: wrap;
  margin-right: var(--gutter-compensation, -0.5rem);
  margin-left: var(--gutter-compensation, -0.5rem);
}

.row.reverse {
  flex-direction: row-reverse;
}

.col.reverse {
  flex-direction: column-reverse;
}

.col-xs,
.col-xs-1,
.col-xs-2,
.col-xs-3,
.col-xs-4,
.col-xs-5,
.col-xs-6,
.col-xs-7,
.col-xs-8,
.col-xs-9,
.col-xs-10,
.col-xs-11,
.col-xs-12,
.col-xs-offset-0,
.col-xs-offset-1,
.col-xs-offset-2,
.col-xs-offset-3,
.col-xs-offset-4,
.col-xs-offset-5,
.col-xs-offset-6,
.col-xs-offset-7,
.col-xs-offset-8,
.col-xs-offset-9,
.col-xs-offset-10,
.col-xs-offset-11,
.col-xs-offset-12 {
  box-sizing: border-box;
  flex: 0 0 auto;
  padding-right: var(--half-gutter-width, 0.5rem);
  padding-left: var(--half-gutter-width, 0.5rem);
}

.col-xs {
  flex-grow: 1;
  flex-basis: 0;
  max-width: 100%;
}

.col-xs-1 {
  flex-basis: 8.33333333%;
  max-width: 8.33333333%;
}

.col-xs-2 {
  flex-basis: 16.66666667%;
  max-width: 16.66666667%;
}

.col-xs-3 {
  flex-basis: 25%;
  max-width: 25%;
}

.col-xs-4 {
  flex-basis: 33.33333333%;
  max-width: 33.33333333%;
}

.col-xs-5 {
  flex-basis: 41.66666667%;
  max-width: 41.66666667%;
}

.col-xs-6 {
  flex-basis: 50%;
  max-width: 50%;
}

.col-xs-7 {
  flex-basis: 58.33333333%;
  max-width: 58.33333333%;
}

.col-xs-8 {
  flex-basis: 66.66666667%;
  max-width: 66.66666667%;
}

.col-xs-9 {
  flex-basis: 75%;
  max-width: 75%;
}

.col-xs-10 {
  flex-basis: 83.33333333%;
  max-width: 83.33333333%;
}

.col-xs-11 {
  flex-basis: 91.66666667%;
  max-width: 91.66666667%;
}

.col-xs-12 {
  flex-basis: 100%;
  max-width: 100%;
}

.col-xs-offset-0 {
  margin-left: 0;
}

.col-xs-offset-1 {
  margin-left: 8.33333333%;
}

.col-xs-offset-2 {
  margin-left: 16.66666667%;
}

.col-xs-offset-3 {
  margin-left: 25%;
}

.col-xs-offset-4 {
  margin-left: 33.33333333%;
}

.col-xs-offset-5 {
  margin-left: 41.66666667%;
}

.col-xs-offset-6 {
  margin-left: 50%;
}

.col-xs-offset-7 {
  margin-left: 58.33333333%;
}

.col-xs-offset-8 {
  margin-left: 66.66666667%;
}

.col-xs-offset-9 {
  margin-left: 75%;
}

.col-xs-offset-10 {
  margin-left: 83.33333333%;
}

.col-xs-offset-11 {
  margin-left: 91.66666667%;
}

.justify-xs-start {
  justify-content: flex-start;
  text-align: start;
}

.justify-xs-center {
  justify-content: center;
  text-align: center;
}

.justify-xs-end {
  justify-content: flex-end;
  text-align: end;
}

.justify-xs-around {
  justify-content: space-around;
}

.justify-xs-between {
  justify-content: space-between;
}

.align-xs-start {
  align-items: flex-start;
}

.align-xs-center {
  align-items: center;
}

.align-xs-end {
  align-items: flex-end;
}

@media screen and (min-width: 768px) {
  .col-sm,
  .col-sm-1,
  .col-sm-2,
  .col-sm-3,
  .col-sm-4,
  .col-sm-5,
  .col-sm-6,
  .col-sm-7,
  .col-sm-8,
  .col-sm-9,
  .col-sm-10,
  .col-sm-11,
  .col-sm-12,
  .col-sm-offset-0,
  .col-sm-offset-1,
  .col-sm-offset-2,
  .col-sm-offset-3,
  .col-sm-offset-4,
  .col-sm-offset-5,
  .col-sm-offset-6,
  .col-sm-offset-7,
  .col-sm-offset-8,
  .col-sm-offset-9,
  .col-sm-offset-10,
  .col-sm-offset-11,
  .col-sm-offset-12 {
    box-sizing: border-box;
    flex: 0 0 auto;
    padding-right: var(--half-gutter-width, 0.5rem);
    padding-left: var(--half-gutter-width, 0.5rem);
  }

  .col-sm {
    flex-grow: 1;
    flex-basis: 0;
    max-width: 100%;
  }

  .col-sm-1 {
    flex-basis: 8.33333333%;
    max-width: 8.33333333%;
  }

  .col-sm-2 {
    flex-basis: 16.66666667%;
    max-width: 16.66666667%;
  }

  .col-sm-3 {
    flex-basis: 25%;
    max-width: 25%;
  }

  .col-sm-4 {
    flex-basis: 33.33333333%;
    max-width: 33.33333333%;
  }

  .col-sm-5 {
    flex-basis: 41.66666667%;
    max-width: 41.66666667%;
  }

  .col-sm-6 {
    flex-basis: 50%;
    max-width: 50%;
  }

  .col-sm-7 {
    flex-basis: 58.33333333%;
    max-width: 58.33333333%;
  }

  .col-sm-8 {
    flex-basis: 66.66666667%;
    max-width: 66.66666667%;
  }

  .col-sm-9 {
    flex-basis: 75%;
    max-width: 75%;
  }

  .col-sm-10 {
    flex-basis: 83.33333333%;
    max-width: 83.33333333%;
  }

  .col-sm-11 {
    flex-basis: 91.66666667%;
    max-width: 91.66666667%;
  }

  .col-sm-12 {
    flex-basis: 100%;
    max-width: 100%;
  }

  .col-sm-offset-0 {
    margin-left: 0;
  }

  .col-sm-offset-1 {
    margin-left: 8.33333333%;
  }

  .col-sm-offset-2 {
    margin-left: 16.66666667%;
  }

  .col-sm-offset-3 {
    margin-left: 25%;
  }

  .col-sm-offset-4 {
    margin-left: 33.33333333%;
  }

  .col-sm-offset-5 {
    margin-left: 41.66666667%;
  }

  .col-sm-offset-6 {
    margin-left: 50%;
  }

  .col-sm-offset-7 {
    margin-left: 58.33333333%;
  }

  .col-sm-offset-8 {
    margin-left: 66.66666667%;
  }

  .col-sm-offset-9 {
    margin-left: 75%;
  }

  .col-sm-offset-10 {
    margin-left: 83.33333333%;
  }

  .col-sm-offset-11 {
    margin-left: 91.66666667%;
  }

  .justify-sm-start {
    justify-content: flex-start;
    text-align: start;
  }

  .justify-sm-center {
    justify-content: center;
    text-align: center;
  }

  .justify-sm-end {
    justify-content: flex-end;
    text-align: end;
  }

  .justify-sm-around {
    justify-content: space-around;
  }

  .justify-sm-between {
    justify-content: space-between;
  }

  .align-sm-start {
    align-items: flex-start;
  }

  .align-sm-center {
    align-items: center;
  }

  .align-sm-end {
    align-items: flex-end;
  }
}

@media screen and (min-width: 1024px) {
  .col-md,
  .col-md-1,
  .col-md-2,
  .col-md-3,
  .col-md-4,
  .col-md-5,
  .col-md-6,
  .col-md-7,
  .col-md-8,
  .col-md-9,
  .col-md-10,
  .col-md-11,
  .col-md-12,
  .col-md-offset-0,
  .col-md-offset-1,
  .col-md-offset-2,
  .col-md-offset-3,
  .col-md-offset-4,
  .col-md-offset-5,
  .col-md-offset-6,
  .col-md-offset-7,
  .col-md-offset-8,
  .col-md-offset-9,
  .col-md-offset-10,
  .col-md-offset-11,
  .col-md-offset-12 {
    box-sizing: border-box;
    flex: 0 0 auto;
    padding-right: var(--half-gutter-width, 0.5rem);
    padding-left: var(--half-gutter-width, 0.5rem);
  }

  .col-md {
    flex-grow: 1;
    flex-basis: 0;
    max-width: 100%;
  }

  .col-md-1 {
    flex-basis: 8.33333333%;
    max-width: 8.33333333%;
  }

  .col-md-2 {
    flex-basis: 16.66666667%;
    max-width: 16.66666667%;
  }

  .col-md-3 {
    flex-basis: 25%;
    max-width: 25%;
  }

  .col-md-4 {
    flex-basis: 33.33333333%;
    max-width: 33.33333333%;
  }

  .col-md-5 {
    flex-basis: 41.66666667%;
    max-width: 41.66666667%;
  }

  .col-md-6 {
    flex-basis: 50%;
    max-width: 50%;
  }

  .col-md-7 {
    flex-basis: 58.33333333%;
    max-width: 58.33333333%;
  }

  .col-md-8 {
    flex-basis: 66.66666667%;
    max-width: 66.66666667%;
  }

  .col-md-9 {
    flex-basis: 75%;
    max-width: 75%;
  }

  .col-md-10 {
    flex-basis: 83.33333333%;
    max-width: 83.33333333%;
  }

  .col-md-11 {
    flex-basis: 91.66666667%;
    max-width: 91.66666667%;
  }

  .col-md-12 {
    flex-basis: 100%;
    max-width: 100%;
  }

  .col-md-offset-0 {
    margin-left: 0;
  }

  .col-md-offset-1 {
    margin-left: 8.33333333%;
  }

  .col-md-offset-2 {
    margin-left: 16.66666667%;
  }

  .col-md-offset-3 {
    margin-left: 25%;
  }

  .col-md-offset-4 {
    margin-left: 33.33333333%;
  }

  .col-md-offset-5 {
    margin-left: 41.66666667%;
  }

  .col-md-offset-6 {
    margin-left: 50%;
  }

  .col-md-offset-7 {
    margin-left: 58.33333333%;
  }

  .col-md-offset-8 {
    margin-left: 66.66666667%;
  }

  .col-md-offset-9 {
    margin-left: 75%;
  }

  .col-md-offset-10 {
    margin-left: 83.33333333%;
  }

  .col-md-offset-11 {
    margin-left: 91.66666667%;
  }

  .justify-md-start {
    justify-content: flex-start;
    text-align: start;
  }

  .justify-md-center {
    justify-content: center;
    text-align: center;
  }

  .justify-md-end {
    justify-content: flex-end;
    text-align: end;
  }

  .justify-md-around {
    justify-content: space-around;
  }

  .justify-md-between {
    justify-content: space-between;
  }

  .align-md-start {
    align-items: flex-start;
  }

  .align-md-center {
    align-items: center;
  }

  .align-md-end {
    align-items: flex-end;
  }
}

@media screen and (min-width: 1280px) {
  .col-lg,
  .col-lg-1,
  .col-lg-2,
  .col-lg-3,
  .col-lg-4,
  .col-lg-5,
  .col-lg-6,
  .col-lg-7,
  .col-lg-8,
  .col-lg-9,
  .col-lg-10,
  .col-lg-11,
  .col-lg-12,
  .col-lg-offset-0,
  .col-lg-offset-1,
  .col-lg-offset-2,
  .col-lg-offset-3,
  .col-lg-offset-4,
  .col-lg-offset-5,
  .col-lg-offset-6,
  .col-lg-offset-7,
  .col-lg-offset-8,
  .col-lg-offset-9,
  .col-lg-offset-10,
  .col-lg-offset-11,
  .col-lg-offset-12 {
    box-sizing: border-box;
    flex: 0 0 auto;
    padding-right: var(--half-gutter-width, 0.5rem);
    padding-left: var(--half-gutter-width, 0.5rem);
  }

  .col-lg {
    flex-grow: 1;
    flex-basis: 0;
    max-width: 100%;
  }

  .col-lg-1 {
    flex-basis: 8.33333333%;
    max-width: 8.33333333%;
  }

  .col-lg-2 {
    flex-basis: 16.66666667%;
    max-width: 16.66666667%;
  }

  .col-lg-3 {
    flex-basis: 25%;
    max-width: 25%;
  }

  .col-lg-4 {
    flex-basis: 33.33333333%;
    max-width: 33.33333333%;
  }

  .col-lg-5 {
    flex-basis: 41.66666667%;
    max-width: 41.66666667%;
  }

  .col-lg-6 {
    flex-basis: 50%;
    max-width: 50%;
  }

  .col-lg-7 {
    flex-basis: 58.33333333%;
    max-width: 58.33333333%;
  }

  .col-lg-8 {
    flex-basis: 66.66666667%;
    max-width: 66.66666667%;
  }

  .col-lg-9 {
    flex-basis: 75%;
    max-width: 75%;
  }

  .col-lg-10 {
    flex-basis: 83.33333333%;
    max-width: 83.33333333%;
  }

  .col-lg-11 {
    flex-basis: 91.66666667%;
    max-width: 91.66666667%;
  }

  .col-lg-12 {
    flex-basis: 100%;
    max-width: 100%;
  }

  .col-lg-offset-0 {
    margin-left: 0;
  }

  .col-lg-offset-1 {
    margin-left: 8.33333333%;
  }

  .col-lg-offset-2 {
    margin-left: 16.66666667%;
  }

  .col-lg-offset-3 {
    margin-left: 25%;
  }

  .col-lg-offset-4 {
    margin-left: 33.33333333%;
  }

  .col-lg-offset-5 {
    margin-left: 41.66666667%;
  }

  .col-lg-offset-6 {
    margin-left: 50%;
  }

  .col-lg-offset-7 {
    margin-left: 58.33333333%;
  }

  .col-lg-offset-8 {
    margin-left: 66.66666667%;
  }

  .col-lg-offset-9 {
    margin-left: 75%;
  }

  .col-lg-offset-10 {
    margin-left: 83.33333333%;
  }

  .col-lg-offset-11 {
    margin-left: 91.66666667%;
  }

  .justify-lg-start {
    justify-content: flex-start;
    text-align: start;
  }

  .justify-lg-center {
    justify-content: center;
    text-align: center;
  }

  .justify-lg-end {
    justify-content: flex-end;
    text-align: end;
  }

  .justify-lg-around {
    justify-content: space-around;
  }

  .justify-lg-between {
    justify-content: space-between;
  }

  .align-lg-start {
    align-items: flex-start;
  }

  .align-lg-center {
    align-items: center;
  }

  .align-lg-end {
    align-items: flex-end;
  }
}

@media screen and (min-width: 1440px) {
  .col-xl,
  .col-xl-1,
  .col-xl-2,
  .col-xl-3,
  .col-xl-4,
  .col-xl-5,
  .col-xl-6,
  .col-xl-7,
  .col-xl-8,
  .col-xl-9,
  .col-xl-10,
  .col-xl-11,
  .col-xl-12,
  .col-xl-offset-0,
  .col-xl-offset-1,
  .col-xl-offset-2,
  .col-xl-offset-3,
  .col-xl-offset-4,
  .col-xl-offset-5,
  .col-xl-offset-6,
  .col-xl-offset-7,
  .col-xl-offset-8,
  .col-xl-offset-9,
  .col-xl-offset-10,
  .col-xl-offset-11,
  .col-xl-offset-12 {
    box-sizing: border-box;
    flex: 0 0 auto;
    padding-right: var(--half-gutter-width, 0.5rem);
    padding-left: var(--half-gutter-width, 0.5rem);
  }

  .col-xl {
    flex-grow: 1;
    flex-basis: 0;
    max-width: 100%;
  }

  .col-xl-1 {
    flex-basis: 8.33333333%;
    max-width: 8.33333333%;
  }

  .col-xl-2 {
    flex-basis: 16.66666667%;
    max-width: 16.66666667%;
  }

  .col-xl-3 {
    flex-basis: 25%;
    max-width: 25%;
  }

  .col-xl-4 {
    flex-basis: 33.33333333%;
    max-width: 33.33333333%;
  }

  .col-xl-5 {
    flex-basis: 41.66666667%;
    max-width: 41.66666667%;
  }

  .col-xl-6 {
    flex-basis: 50%;
    max-width: 50%;
  }

  .col-xl-7 {
    flex-basis: 58.33333333%;
    max-width: 58.33333333%;
  }

  .col-xl-8 {
    flex-basis: 66.66666667%;
    max-width: 66.66666667%;
  }

  .col-xl-9 {
    flex-basis: 75%;
    max-width: 75%;
  }

  .col-xl-10 {
    flex-basis: 83.33333333%;
    max-width: 83.33333333%;
  }

  .col-xl-11 {
    flex-basis: 91.66666667%;
    max-width: 91.66666667%;
  }

  .col-xl-12 {
    flex-basis: 100%;
    max-width: 100%;
  }

  .col-xl-offset-0 {
    margin-left: 0;
  }

  .col-xl-offset-1 {
    margin-left: 8.33333333%;
  }

  .col-xl-offset-2 {
    margin-left: 16.66666667%;
  }

  .col-xl-offset-3 {
    margin-left: 25%;
  }

  .col-xl-offset-4 {
    margin-left: 33.33333333%;
  }

  .col-xl-offset-5 {
    margin-left: 41.66666667%;
  }

  .col-xl-offset-6 {
    margin-left: 50%;
  }

  .col-xl-offset-7 {
    margin-left: 58.33333333%;
  }

  .col-xl-offset-8 {
    margin-left: 66.66666667%;
  }

  .col-xl-offset-9 {
    margin-left: 75%;
  }

  .col-xl-offset-10 {
    margin-left: 83.33333333%;
  }

  .col-xl-offset-11 {
    margin-left: 91.66666667%;
  }

  .justify-xl-start {
    justify-content: flex-start;
    text-align: start;
  }

  .justify-xl-center {
    justify-content: center;
    text-align: center;
  }

  .justify-xl-end {
    justify-content: flex-end;
    text-align: end;
  }

  .justify-xl-around {
    justify-content: space-around;
  }

  .justify-xl-between {
    justify-content: space-between;
  }

  .align-xl-start {
    align-items: flex-start;
  }

  .align-xl-center {
    align-items: center;
  }

  .align-xl-end {
    align-items: flex-end;
  }
}
