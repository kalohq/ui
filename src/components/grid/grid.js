/* @flow */
import React from 'react';
import styled from 'react-emotion';
import {toNumber} from 'lodash';

import {Box, Flex} from '../layout';

/**
 * Some low level grid components
 */
export const Grid = styled(Flex)`
  width: 100%;
  flex-direction: column;
  max-width: ${props => props.theme.grid.gridMaxWidth}px;
  margin-left: auto;
  margin-right: auto;
`;

Grid.displayName = 'Grid';

type TRowProps = {
  children?: any,
  gutter?: 'extra-small' | 'small' | 'medium' | 'large' | 'extra-large',
  collapse?: boolean,
  spacing?: 'extra-small' | 'small' | 'medium' | 'large' | 'extra-large',
};

export function Row(props: TRowProps) {
  const {
    children,
    spacing = 'medium',
    gutter,
    collapse = false,
    ...otherProps
  } = props;

  return (
    <Flex
      width="100%"
      flexWrap="wrap"
      paddingLeft={gutter}
      paddingRight={gutter}
      {...otherProps}
    >
      {React.Children.map(
        children,
        child =>
          child
            ? React.cloneElement(child, {
                paddingLeft: !collapse ? spacing : null,
                paddingRight: !collapse ? spacing : null,
              })
            : null
      )}
    </Flex>
  );
}

type TColumnProps = {
  columns?: number,
  children?: any,
  spacing?: 'small',
};

export function Column(props: TColumnProps) {
  const {columns = 12, children, ...otherProps} = props;

  const _width = `${100 / 12 * toNumber(columns)}%`;

  return (
    <Box flexDirection="column" width={_width} {...otherProps}>
      {children}
    </Box>
  );
}
