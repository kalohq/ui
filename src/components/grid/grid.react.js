/* @flow */
import React from 'react';
import cx from 'classnames';
import {parseStyleProps, cleanProps} from '../../utils/style';

import styles from './grid.css';

/**
 * Grid Container
 */

type TGridProps = {
  children?: any,
  fluid?: boolean,
  className?: string | Object,
};

export function Grid(props: TGridProps) {
  const {children, fluid, className, ...otherProps} = props;

  const _classNames = cx(
    {
      [styles.container]: true,
      [styles.fluid]: fluid,
    },
    className
  );

  return (
    <Spacer className={_classNames} {...otherProps}>
      {children}
    </Spacer>
  );
}

/**
 * Generic Spacer
 */

type TSpacerProps = {
  is?: string,
  children?: any,
  className?: string | Object,
};

const Spacer = (originalProps: TSpacerProps) => {
  const {
    is: Element = 'div',
    children,
    className,
    ...otherProps
  } = originalProps;

  const {props, style} = parseStyleProps({
    ...otherProps,
  });

  const cleanedProps = cleanProps(props);

  return (
    <Element className={className} {...cleanedProps} css={style}>
      {children}
    </Element>
  );
};

/**
 * Row
 */

type TRowProps = {
  children?: any,
  gutters?:
    | 'extra-small'
    | 'small'
    | 'medium'
    | 'large'
    | 'extra-large'
    | 'none',
  collapse?: boolean,
  reverse?: boolean,
  alignItems?: Array<*> | string,
  justifyContent?: Array<*> | string,
  className?: string | Object,
};

export function Row(props: TRowProps) {
  const {
    children,
    reverse,
    alignItems = [],
    justifyContent = [],
    className,
    ...otherProps
  } = props;

  const [
    alignItemsXS,
    alignItemsSM,
    alignItemsMD,
    alignItemsLG,
  ] = (Array.isArray(alignItems) ? alignItems : [alignItems]).map(val =>
    val.replace('flex-', '')
  );

  const [
    justifyContentXS,
    justifyContentSM,
    justifyContentMD,
    justifyContentLG,
  ] = (Array.isArray(justifyContent) ? justifyContent : [justifyContent]).map(
    val => val.replace('flex-', '').replace('space-', '')
  );

  const _classNames = cx(
    {
      [styles.row]: true,
      [styles.reverse]: reverse,
      [styles[`align-xs-${alignItemsXS}`]]: alignItemsXS,
      [styles[`align-sm-${alignItemsSM}`]]: alignItemsSM,
      [styles[`align-md-${alignItemsMD}`]]: alignItemsMD,
      [styles[`align-lg-${alignItemsLG}`]]: alignItemsLG,
      [styles[`justify-xs-${justifyContentXS}`]]: justifyContentXS,
      [styles[`justify-sm-${justifyContentSM}`]]: justifyContentSM,
      [styles[`justify-md-${justifyContentMD}`]]: justifyContentMD,
      [styles[`justify-lg-${justifyContentLG}`]]: justifyContentLG,
    },
    className
  );

  return (
    <Spacer className={_classNames} {...otherProps}>
      {children}
    </Spacer>
  );
}

/**
 * Column
 */

type TColumnProps = {
  columns?: Array<*> | number,
  children?: any,
  largeColumn?: number,
  className?: string | Object,
};

export function Column(props: TColumnProps) {
  const {columns = [12], children, className, ...otherProps} = props;

  const [columnXS, columnSM, columnMD, columnLG, columnXL] = Array.isArray(
    columns
  )
    ? columns
    : [columns];

  const _classNames = cx(
    {
      [styles[`col-xs-${String(columnXS)}`]]: columnXS,
      [styles[`col-sm-${String(columnSM)}`]]: columnSM,
      [styles[`col-md-${String(columnMD)}`]]: columnMD,
      [styles[`col-lg-${String(columnLG)}`]]: columnLG,
      [styles[`col-xl-${String(columnXL)}`]]: columnXL,
    },
    className
  );

  return (
    <Spacer className={_classNames} {...otherProps}>
      {children}
    </Spacer>
  );
}
