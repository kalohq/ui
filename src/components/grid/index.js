/* @flow */
export {Grid, Row, Column} from './grid.react';
