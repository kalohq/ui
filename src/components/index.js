/**
 * Typography
 */
export {default as H1} from './h1';
export {default as H2} from './h2';
export {default as H3} from './h3';
export {default as H4} from './h4';
export {default as Heading} from './heading';
export {default as Text} from './text';
export {default as HelperText} from './helper-text';

/**
 * Layout
 */
export * from './skeleton';
export {Box, Flex, Block, Inline, InlineFlex, A, UIBase} from './layout';
export {default as Paper, PaperToolbar} from './paper';
export {default as PaperMenu, PaperMenuItem} from './paper-menu';
export {default as DropMenu, DropMenuItem} from './drop-menu';
export {Grid, Row, Column} from './grid';

/**
 * Form
 */
export {default as Button} from './button';
export {default as ButtonDropdown} from './button-dropdown';
export {default as ButtonGroup} from './button-group';
export {default as Checkbox} from './checkbox';
export {default as Field} from './field';
export {default as FieldHint} from './field-hint';
export {default as FieldLabel} from './field-label';
export {default as FieldRow} from './field-row';
export {default as FieldValidation} from './field-validation';
export {default as FieldValidations} from './field-validations';
export {default as Fieldset} from './fieldset';
export {default as Form} from './form';
export {default as Input} from './input';
export {default as ImageInput} from './image-input';
export {Option as SelectOption} from './select';
export {default as Radio} from './radio';
export {default as Select} from './select';
export {default as Textarea} from './textarea';
export {
  default as ToggleButton,
  HEIGHT as TOGGLE_BUTTON_HEIGHT,
} from './toggle-button';
export {default as SeamlessButton} from './seamless-button';

/**
 * Table
 */
export {
  default as Table,
  TableHead,
  TableBody,
  TableRow,
  TableCell,
} from './table';

/**
 * Other
 */
export {default as Alert, AlertPopover} from './alert';
export {default as Icon} from './icon';
export {default as IconSymbols} from './icon-symbols';
export {default as LoadingSpinner} from './loading-spinner';
export {default as Sticky} from './sticky';
export {default as StarRating} from './star-rating';
export {default as Lozenge} from './lozenge';
export {default as Tag} from './tag';
export {default as SkillTag} from './skill-tag';
export {default as TagGroup} from './tag-group';
export {default as Avatar} from './avatar';
export {default as AvatarGroup} from './avatar-group';
export {default as Modal} from './modal';
export {Tab, Tabs} from './tabs';
export {default as FeatureCalloutCard} from './feature-callout-card';
export {default as Callout} from './callout';

/**
 * Theme
 */
export {default as theme} from './theme';
