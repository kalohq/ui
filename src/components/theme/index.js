/* @flow */
export {default} from './theme';
