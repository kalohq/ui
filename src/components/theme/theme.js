/* @flow */

import tokens from '../../design-tokens/tokens.theme.js';

/**
 * ThemeProvider Kalo Theme
 * 
 * This is the core CSS-in-JS theme that is passed in to the
 * global ThemeProvider. It is mainly composed of our core
 * design tokens.
 */

const theme = {
  space: [0, 8, 16, 32, 64],
  breakpoints: [],
  ...tokens,
};

export default theme;
