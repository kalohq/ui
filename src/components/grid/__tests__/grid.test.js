/* @flow */
/* eslint-env jest */
import * as React from 'react';
import * as emotion from 'emotion';
import {createSerializer} from 'jest-emotion';
import renderer from 'react-test-renderer';
import {ThemeProvider} from 'emotion-theming';

import theme from 'components/theme';
import {Grid, Row, Column} from 'components/grid';

expect.addSnapshotSerializer(createSerializer(emotion));

describe('components/grid', () => {
  const create = (props = {}) =>
    renderer
      .create(
        <ThemeProvider theme={theme}>
          <Grid {...props}>
            <Row>
              <Column columns={2} />
              <Column columns={10} />
            </Row>
          </Grid>
        </ThemeProvider>
      )
      .toJSON();

  test('should render shallow component ok', () => {
    const element = create();
    expect(element).toMatchSnapshot();
  });
});
