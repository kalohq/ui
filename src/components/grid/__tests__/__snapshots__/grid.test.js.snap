// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/grid should render shallow component ok 1`] = `
.emotion-7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 100%;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  max-width: 1200px;
  margin-left: auto;
  margin-right: auto;
}

.emotion-5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 100%;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
}

.emotion-1 {
  position: relative;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-align-content: flex-start;
  -ms-flex-line-pack: flex-start;
  align-content: flex-start;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 16.666666666666668%;
  padding-left: 16px;
  padding-right: 16px;
}

.emotion-3 {
  position: relative;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-align-content: flex-start;
  -ms-flex-line-pack: flex-start;
  align-content: flex-start;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 83.33333333333334%;
  padding-left: 16px;
  padding-right: 16px;
}

<div
  className="emotion-6 emotion-7"
  style={Object {}}
>
  <div
    className="emotion-4 emotion-5"
    style={Object {}}
  >
    <div
      className="emotion-0 emotion-1"
      style={Object {}}
    />
    <div
      className="emotion-0 emotion-3"
      style={Object {}}
    />
  </div>
</div>
`;
